service.go

```diff
@@ -4,75 +4,115 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"sync"
+	"time"
+
 	"github.com/Financial-Times/tme-reader/tmereader"
 	log "github.com/Sirupsen/logrus"
 	"github.com/boltdb/bolt"
-	"github.com/pborman/uuid"
-	"sync"
-	"time"
 )
 
 const (
-	cacheBucket  = "org"
-	uppAuthority = "http://api.ft.com/system/FT-UPP"
-	tmeAuthority = "http://api.ft.com/system/FT-TME"
+	cacheBucket = "org"
 )
 
 type orgsService interface {
-	getOrgs() ([]orgLink, bool)
+	getOrgs() ([]orgLink, error)
 	getOrgByUUID(uuid string) (org, bool, error)
 	isInitialised() bool
+	isDataLoaded() bool
 	shutdown() error
+	orgCount() (int, error)
+	orgIds() ([]orgUUID, error)
+	orgReload() error
 }
 
 type orgServiceImpl struct {
+	sync.RWMutex
 	repository    tmereader.Repository
 	baseURL       string
-	orgLinks      []orgLink
 	taxonomyName  string
 	maxTmeRecords int
 	initialised   bool
+	dataLoaded    bool
 	cacheFileName string
 	db            *bolt.DB
 }
 
 func newOrgService(repo tmereader.Repository, baseURL string, taxonomyName string, maxTmeRecords int, cacheFileName string) orgsService {
-	s := &orgServiceImpl{repository: repo, baseURL: baseURL, taxonomyName: taxonomyName, maxTmeRecords: maxTmeRecords, initialised: false, cacheFileName: cacheFileName}
+	s := &orgServiceImpl{repository: repo, baseURL: baseURL, taxonomyName: taxonomyName, maxTmeRecords: maxTmeRecords, initialised: true, dataLoaded: false, cacheFileName: cacheFileName}
 	go func(service *orgServiceImpl) {
 		err := service.init()
 		if err != nil {
 			log.Errorf("Error while creating OrgService: [%v]", err.Error())
 		}
-		service.initialised = true
+		s.setDataLoaded(true)
 	}(s)
 	return s
 }
 
 func (s *orgServiceImpl) isInitialised() bool {
+	s.RLock()
+	defer s.RUnlock()
 	return s.initialised
 }
 
+func (s *orgServiceImpl) setInitialised(val bool) {
+	s.Lock()
+	defer s.Unlock()
+	s.initialised = val
+}
+
+func (s *orgServiceImpl) isDataLoaded() bool {
+	s.RLock()
+	defer s.RUnlock()
+	return s.dataLoaded
+}
+
+func (s *orgServiceImpl) setDataLoaded(val bool) {
+	s.Lock()
+	defer s.Unlock()
+	s.dataLoaded = val
+}
+
 func (s *orgServiceImpl) shutdown() error {
 	if s.db == nil {
```

Comment:
Might want to lock the service to shut it down. Then again if shutdown only called on app quitting then it shouldn't be an issue.